core/src/main/java/io/questdb/cutlass/line/tcp/IOTableUpdateDetailsPool.java

```diff
@@ -0,0 +1,135 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.cutlass.line.tcp;
+
+import io.questdb.log.Log;
+import io.questdb.log.LogFactory;
+import io.questdb.std.ConcurrentHashMap;
+import io.questdb.std.Misc;
+import io.questdb.std.Unsafe;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.function.Function;
+
+class IOTableUpdateDetailsPool implements Closeable {
+    private static final int ABASE;
+    private static final int ASHIFT;
+    private final Log LOG = LogFactory.getLog(IOTableUpdateDetailsPool.class);
+    private final Function<CharSequence, ? extends TableUpdateDetails[]> createNewTableUpdateDetails;
+    private final ConcurrentHashMap<TableUpdateDetails[]> pool = new ConcurrentHashMap<>();
```

Comment:
ConcurrentHashMap can be case-insensitive

Reply:
I left the pool as case sensitive because it is not simple to keep it case insensitive while the maps which matched against ILP lines are in `LineTcpNetworkIOJob.tableUpdateDetailsUtf8` and are case sensitive.

I was not trivial to turn `ByteCharSequenceObjHashMap` into case insensitive mode without performance impact for me so I left WAL resolution to be case sensitive. So if you send lines

```
up out=1.0 631150000000000000
UP out=3.0 631160000000000000
```

Then 2 `TableUpdateDetails` are created and used to process the data even from single connection. This is how it works at the moment before this change.

Reply:
gotcha